### ISSUE TITLE
Feature Request: Would like to put `typeof` in a local variable then take its `sizeof`

### ISSUE THREAD

**Issue by mattrberry** [crystal-lang/crystal#12274]
Currently, this isn't supported and the docs accurately reflect the behavior, saying

> *type* must be a constant or `typeof()` expression. It cannot be evaluated

I think it'd be nice to be able to put the result of a `typeof` in a local variable then check the size of it. No idea of the level of complexity this adds to the compiler. Maybe worth it if it's easy, probably not if it's challenging.

```crystal
puts sizeof(typeof(1)) # => 4
t = typeof(1) # => Int32
puts sizeof(t) # => Error: unexpected token: "t"
```

**Comment by asterite** [comment]
Could you expand a bit the intended use case? Why not do sizeof(typeof(t))?

**Comment by mattrberry** [comment]
It's super minor. I have a scope where I already have the typeof stored in a local variable, and I wanted to take the size of that type without writing typeof again. Super super minor

**Comment by asterite** [comment]
It's a bit strange because `t` in the example above if a runtime value, but `sizeof` operates at compile-time. These shouldn't be mixed.

That said... does this help?

```crystal
def runtime_sizeof(x : T.class) forall T
  sizeof(T)
end

puts sizeof(typeof(1)) # => 4
t = typeof(1)          # => Int32
puts runtime_sizeof(t) # => 4

puts sizeof(typeof("hello")) # => 8
u = typeof("hello")          # => Int32
puts runtime_sizeof(u)       # => 8
```

**Comment by mattrberry** [comment]
Makes sense! I was thinking maybe Crystal could infer that the `t` above was bound to `typeof` which is known at compile-time, but totally understandable that the indirection there makes that tricky!
